torch/_meta_registrations.py

```diff
@@ -190,6 +192,35 @@ def meta_bernoulli(self, *, generator=None, out):
     return out
 
 
+@torch.library.impl(meta_lib, "convolution_overrideable")
+def meta_conv2d(
+    input_tensor: torch.Tensor,
+    weight: torch.Tensor,
+    bias: torch.Tensor,
+    stride: List[int],
+    padding: List[int],
+    dilation: List[int],
+    *args,
+):
+    def calc_conv_return_shape(dim, kernel_size, stride, padding, dilation):
```

Comment:
it may be good to still have such functions public in torch.nn.utils

Reply:
Yup, I was considering that too. Thanks for the feedback, will implement

Reply:
I'm not sure we want to add such things to nn.utils. If we add these for all ops, this is going to be quite a bit of duplication.
Also this formula is already available in the doc.

cc @jbschlosser what do you think?

Reply:
If many ops are reusing this formula (e.g. convs, max-pool etc), I think it'd be de-duplication, right?

Reply:
I'd vote against any additional public functions providing the formulas. Also, longer term, I'd expect that this meta registration will not be needed if convolution is refactored to use structured kernels - @anjali411 is that correct?

Reply:
> If many ops are reusing this formula (e.g. convs, max-pool etc), I think it'd be de-duplication, right?

The problem is that each of these ops have different rarely-used arguments (output side, dilation, group, etc) that makes de-duplication hard.

Reply:
@jbschlosser yes indeed! This is only a workaround for ops that are not structured.